applications/asset_tracker/src/main.c

```diff
@@ -1316,24 +1483,76 @@ void main(void)
 	}
 
 	work_init();
-	modem_configure();
+
+lte_connect:
+		ret = modem_configure();
+		if (ret) {
+			LOG_WRN("Failed to establish LTE connection.");
+			LOG_WRN("Will retry in %d seconds.",
+					CONFIG_CLOUD_CONNECT_RETRY_DELAY);
+			k_sleep(K_SECONDS(CONFIG_CLOUD_CONNECT_RETRY_DELAY));
+			goto lte_connect;
+		}
+
+#if defined(CONFIG_LWM2M_CARRIER)
+	LOG_INF("Waiting for LWM2M carrier to complete initialization...");
+	k_sem_take(&cloud_ready_to_connect, K_FOREVER);
+#endif
+
 connect:
+
+	/* Ensure no data can be sent to cloud before connction is established.
+	 */
+	atomic_set(&cloud_association, CLOUD_ASSOCIATION_STATE_INIT);
+
+	/* Carrier FOTA happens in the background, but it uses the TLS socket
+	 * that cloud also would use. The carrier library will reboot the device
+	 * when the FOTA is done.
+	 */
+	if (atomic_get(&carrier_requested_disconnect)) {
+		return;
+	}
+
+	atomic_inc(&cloud_connect_attempts);
+
+	/* Check if max cloud connect retry count is exceeded. */
+	if (atomic_get(&cloud_connect_attempts) >
+		CONFIG_CLOUD_CONNECT_COUNT_MAX) {
+		LOG_ERR("The max cloud connection attempt count exceeded.");
+		cloud_error_handler(-ETIMEDOUT);
+	}
+
+	if (reconnecting) {
+		reconnecting = false;
+		LOG_INF("Attempting reconnect in %d seconds...",
+				reconnect_delay_s);
+		k_delayed_work_cancel(&cloud_reboot_work);
+		k_sleep(K_SECONDS(reconnect_delay_s));
+		reconnect_delay_s = CONFIG_CLOUD_CONNECT_RETRY_DELAY;
+	}
+
+	LOG_INF("Connecting to cloud, attempt %d of %d",
+	       atomic_get(&cloud_connect_attempts),
+		   CONFIG_CLOUD_CONNECT_COUNT_MAX);
+
+	/* Attempt cloud connection. */
 	ret = cloud_connect(cloud_backend);
 	if (ret != CLOUD_CONNECT_RES_SUCCESS) {
+		/* Will not return from this function. */
+		/* If the connect fails here, it is likely */
+		/* that user intervention is required. */
```

Comment:
Nit: Comment should be multiline inste of multiple one-liners

Reply:
for some reason i thought the CI complained about multi-line comments *not* being formatted this way (into one-liners).  am i remembering that incorrectly? 